type/__package_pip/explorer/state

```diff
@@ -41,7 +41,8 @@ if ! command -v "$pip" >/dev/null 2>&1; then
     echo absent
 else
 
-    if "$pip" freeze | grep -i -q "^$name=="; then
+    # In `pip freeze` output packages installed from URL are shown as `$package @ $url`.
+    if "$pip" freeze | grep -Eiq "^$name(==| @ )"; then
```

Comment:
Hmm, it doesn't seem to do that for me?!

Which version of `pip` are you using?

Reply:
```
$ pip3 -V
pip 22.0.2 from /usr/lib/python3/dist-packages/pip (python 3.10)
```

:thinking: 

OK, I'll take a look with older pips later.

Reply:
@sideeffect42 what are your py/pip versions?

Reply:
WTF?!

```console
$ python3.10 -m venv testvenv
$ . testvenv/bin/activate
$ (testvenv) $ pip --version
pip 22.0.4 from /private/tmp/testvenv/lib/python3.10/site-packages/pip (python 3.10)
$ (testvenv) $ pip install git+https://github.com/systemd/mkosi.git
Collecting git+https://github.com/systemd/mkosi.git
  Cloning https://github.com/systemd/mkosi.git to /private/var/folders/hm/1zn8rd0d3jgbk342zbxl9sww0000gn/T/pip-req-build-806xtmxf
  Running command git clone --filter=blob:none --quiet https://github.com/systemd/mkosi.git /private/var/folders/hm/1zn8rd0d3jgbk342zbxl9sww0000gn/T/pip-req-build-806xtmxf
  Resolved https://github.com/systemd/mkosi.git to commit 5ab16cda00f5b24848b427ddd28ad62c822079ff
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: mkosi
  Building wheel for mkosi (pyproject.toml) ... done
  Created wheel for mkosi: filename=mkosi-12-py3-none-any.whl size=133103 sha256=fb4d8cc2d1cd190240f4c7ee4b61c53595c4daa8f2a41fc5002dad10bcd629b8
  Stored in directory: /private/var/folders/hm/1zn8rd0d3jgbk342zbxl9sww0000gn/T/pip-ephem-wheel-cache-szif8kna/wheels/00/66/dd/57eaa83f3cb1e06860e210f0e21b743b328fa9e76041a88e5b
Successfully built mkosi
Installing collected packages: mkosi
Successfully installed mkosi-12
$ (testvenv) $ pip install git+https://github.com/cdist-community/cdist-core.git
Collecting git+https://github.com/cdist-community/cdist-core.git
  Cloning https://github.com/cdist-community/cdist-core.git to /private/var/folders/hm/1zn8rd0d3jgbk342zbxl9sww0000gn/T/pip-req-build-ou6sgobb
  Running command git clone --filter=blob:none --quiet https://github.com/cdist-community/cdist-core.git /private/var/folders/hm/1zn8rd0d3jgbk342zbxl9sww0000gn/T/pip-req-build-ou6sgobb
  Resolved https://github.com/cdist-community/cdist-core.git to commit 7f18b568f627bfb926a0f58396726cd9a9bb8b76
  Running command git submodule update --init --recursive -q
  Preparing metadata (setup.py) ... done
Using legacy 'setup.py install' for cdist, since package 'wheel' is not installed.
Installing collected packages: cdist
  Running setup.py install for cdist ... done
Successfully installed cdist-6.9.7-131-g7f18b568
$ (testvenv) $ pip freeze
cdist===6.9.7-131-g7f18b568
mkosi @ git+https://github.com/systemd/mkosi.git@5ab16cda00f5b24848b427ddd28ad62c822079ff
```

Reply:
To make things more confusing:

```console
$ (testvenv) $ pip list --format=freeze
cdist==6.9.7-131-g7f18b568
mkosi==12
pip==22.0.4
setuptools==58.1.0
```

Reply:
Hmm, after studying the source code I found out that:

1) `pip freeze` will not show `distribute`, `setuptools`, `wheel`, `pip` unless `--all` is used.
2) `pip list --format=freeze` output will always print `{name}=={version}` pairs ([src](https://github.com/pypa/pip/blob/main/src/pip/_internal/commands/list.py#L278)) while `pip freeze` prints output for a `requirements.txt` ([src](https://github.com/pypa/pip/blob/main/src/pip/_internal/commands/freeze.py#L87)).

So I think `pip list --format=freeze` (supported with `pip>=9.0.0`) actually produces the output we want.

I think we should also throw in a `--disable-pip-version-check` while at it.